diff -ruN boost_1_49_0-boot/boost/asio/detail/fenced_block.hpp boost_1_49_0-patched/boost/asio/detail/fenced_block.hpp
--- boost_1_49_0-boot/boost/asio/detail/fenced_block.hpp	2012-01-15 14:46:25.000000000 +0100
+++ boost_1_49_0-patched/boost/asio/detail/fenced_block.hpp	2012-06-27 19:19:06.069571660 +0200
@@ -25,7 +25,7 @@
 # include <boost/asio/detail/macos_fenced_block.hpp>
 #elif defined(__sun)
 # include <boost/asio/detail/solaris_fenced_block.hpp>
-#elif defined(__GNUC__) && defined(__arm__)
+#elif defined(__GNUC__) && defined(__arm__) && !defined(__thumb__)
 # include <boost/asio/detail/gcc_arm_fenced_block.hpp>
 #elif defined(__GNUC__) && (defined(__hppa) || defined(__hppa__))
 # include <boost/asio/detail/gcc_hppa_fenced_block.hpp>
@@ -34,7 +34,8 @@
 #elif defined(__GNUC__) \
   && ((__GNUC__ == 4 && __GNUC_MINOR__ >= 1) || (__GNUC__ > 4)) \
   && !defined(__INTEL_COMPILER) && !defined(__ICL) \
-  && !defined(__ICC) && !defined(__ECC) && !defined(__PATHSCALE__)
+  && !defined(__ICC) && !defined(__ECC) && !defined(__PATHSCALE__) \
+  && !defined(ANDROID) && !defined(__ANDROID__)
 # include <boost/asio/detail/gcc_sync_fenced_block.hpp>
 #elif defined(BOOST_WINDOWS) && !defined(UNDER_CE)
 # include <boost/asio/detail/win_fenced_block.hpp>
@@ -54,7 +55,7 @@
 typedef macos_fenced_block fenced_block;
 #elif defined(__sun)
 typedef solaris_fenced_block fenced_block;
-#elif defined(__GNUC__) && defined(__arm__)
+#elif defined(__GNUC__) && defined(__arm__) && !defined(__thumb__)
 typedef gcc_arm_fenced_block fenced_block;
 #elif defined(__GNUC__) && (defined(__hppa) || defined(__hppa__))
 typedef gcc_hppa_fenced_block fenced_block;
@@ -63,7 +64,8 @@
 #elif defined(__GNUC__) \
   && ((__GNUC__ == 4 && __GNUC_MINOR__ >= 1) || (__GNUC__ > 4)) \
   && !defined(__INTEL_COMPILER) && !defined(__ICL) \
-  && !defined(__ICC) && !defined(__ECC) && !defined(__PATHSCALE__)
+  && !defined(__ICC) && !defined(__ECC) && !defined(__PATHSCALE__) \
+  && !defined(ANDROID) && !defined(__ANDROID__)
 typedef gcc_sync_fenced_block fenced_block;
 #elif defined(BOOST_WINDOWS) && !defined(UNDER_CE)
 typedef win_fenced_block fenced_block;
diff -ruN boost_1_49_0-boot/boost/asio/detail/socket_types.hpp boost_1_49_0-patched/boost/asio/detail/socket_types.hpp
--- boost_1_49_0-boot/boost/asio/detail/socket_types.hpp	2012-01-15 14:46:25.000000000 +0100
+++ boost_1_49_0-patched/boost/asio/detail/socket_types.hpp	2012-06-27 19:19:01.279562338 +0200
@@ -123,7 +123,12 @@
 typedef int socket_type;
 const int invalid_socket = -1;
 const int socket_error_retval = -1;
+// @Moss - Some platforms do not define it (Android)
+#if defined(INET_ADDRSTRLEN)
 const int max_addr_v4_str_len = INET_ADDRSTRLEN;
+#else // defined(INET_ADDRSTRLEN)
+const int max_addr_v4_str_len = 16;
+#endif // defined(INET_ADDRSTRLEN)
 #if defined(INET6_ADDRSTRLEN)
 const int max_addr_v6_str_len = INET6_ADDRSTRLEN + 1 + IF_NAMESIZE;
 #else // defined(INET6_ADDRSTRLEN)
diff -ruN boost_1_49_0-boot/boost/asio/ip/impl/address_v6.ipp boost_1_49_0-patched/boost/asio/ip/impl/address_v6.ipp
--- boost_1_49_0-boot/boost/asio/ip/impl/address_v6.ipp	2012-01-15 14:46:25.000000000 +0100
+++ boost_1_49_0-patched/boost/asio/ip/impl/address_v6.ipp	2012-06-27 19:19:11.029581297 +0200
@@ -11,6 +11,23 @@
 #ifndef BOOST_ASIO_IP_IMPL_ADDRESS_V6_IPP
 #define BOOST_ASIO_IP_IMPL_ADDRESS_V6_IPP
 
+// @Moss - Define IPv6 macros
+#if !defined(IN6_IS_ADDR_MULTICAST) 
+#define IN6_IS_ADDR_MULTICAST(a) (((__const uint8_t *) (a))[0] == 0xff)
+#endif
+
+#if !defined(IN6_IS_ADDR_MC_NODELOCAL)
+#define IN6_IS_ADDR_MC_NODELOCAL(a) \
+        (IN6_IS_ADDR_MULTICAST(a) \
+         && ((((__const uint8_t *) (a))[1] & 0xf) == 0x1))
+#endif
+
+#if !defined(IN6_IS_ADDR_MC_GLOBAL)
+#define IN6_IS_ADDR_MC_GLOBAL(a) \
+        (IN6_IS_ADDR_MULTICAST(a) \
+         && ((((__const uint8_t *) (a))[1] & 0xf) == 0xe))
+#endif
+
 #if defined(_MSC_VER) && (_MSC_VER >= 1200)
 # pragma once
 #endif // defined(_MSC_VER) && (_MSC_VER >= 1200)
diff -ruN boost_1_49_0-boot/boost/config/user.hpp boost_1_49_0-patched/boost/config/user.hpp
--- boost_1_49_0-boot/boost/config/user.hpp	2004-01-10 13:10:00.000000000 +0100
+++ boost_1_49_0-patched/boost/config/user.hpp	2012-06-27 19:18:46.129532736 +0200
@@ -13,6 +13,11 @@
 //  configuration policy:
 //
 
+// Android defines
+#define __arm__ 1
+#define _REENTRANT 1
+#define _GLIBCXX__PTHREADS 1
+
 // define this to locate a compiler config file:
 // #define BOOST_COMPILER_CONFIG <myheader>
 
diff -ruN boost_1_49_0-boot/boost/detail/endian.hpp boost_1_49_0-patched/boost/detail/endian.hpp
--- boost_1_49_0-boot/boost/detail/endian.hpp	2011-03-29 23:58:48.000000000 +0200
+++ boost_1_49_0-patched/boost/detail/endian.hpp	2012-06-27 19:18:39.359519453 +0200
@@ -31,7 +31,7 @@
 // GNU libc offers the helpful header <endian.h> which defines
 // __BYTE_ORDER
 
-#if defined (__GLIBC__)
+#if defined (__GLIBC__) || defined(ANDROID)
 # include <endian.h>
 # if (__BYTE_ORDER == __LITTLE_ENDIAN)
 #  define BOOST_LITTLE_ENDIAN
diff -ruN boost_1_49_0-boot/boost/interprocess/detail/workaround.hpp boost_1_49_0-patched/boost/interprocess/detail/workaround.hpp
--- boost_1_49_0-boot/boost/interprocess/detail/workaround.hpp	2011-12-26 18:21:36.000000000 +0100
+++ boost_1_49_0-patched/boost/interprocess/detail/workaround.hpp	2012-06-27 19:18:52.909546004 +0200
@@ -64,7 +64,7 @@
    #endif
 
    //Check for XSI shared memory objects. They are available in nearly all UNIX platforms
-   #if !defined(__QNXNTO__)
+   #if !defined(__QNXNTO__) && !defined(ANDROID)
    # define BOOST_INTERPROCESS_XSI_SHARED_MEMORY_OBJECTS
    #endif
 
diff -ruN boost_1_49_0-boot/libs/filesystem/v2/src/v2_operations.cpp boost_1_49_0-patched/libs/filesystem/v2/src/v2_operations.cpp
--- boost_1_49_0-boot/libs/filesystem/v2/src/v2_operations.cpp	2012-01-15 20:22:27.000000000 +0100
+++ boost_1_49_0-patched/libs/filesystem/v2/src/v2_operations.cpp	2012-06-27 19:19:15.239589462 +0200
@@ -60,13 +60,15 @@
 
 # else // BOOST_POSIX_API
 #   include <sys/types.h>
-#   if !defined(__APPLE__) && !defined(__OpenBSD__)
+#   if !defined(__APPLE__) && !defined(__OpenBSD__) && !defined(__ANDROID__) && !defined(ANDROID)
 #     include <sys/statvfs.h>
 #     define BOOST_STATVFS statvfs
 #     define BOOST_STATVFS_F_FRSIZE vfs.f_frsize
 #   else
 #ifdef __OpenBSD__
 #     include <sys/param.h>
+#elif defined(__ANDROID__) || defined(ANDROID) // @Moss - Android messes up a bit with some headers, this one is the correct one :D
+#     include <sys/vfs.h>
 #endif
 #     include <sys/mount.h>
 #     define BOOST_STATVFS statfs
@@ -1267,7 +1269,11 @@
         if ( max == 0 )
         {
           errno = 0;
+#     ifdef __ANDROID__ || ANDROID
+          long tmp = 4096;
+#     else
           long tmp = ::pathconf( "/", _PC_NAME_MAX );
+#     endif
           if ( tmp < 0 )
           {
             if ( errno == 0 ) // indeterminate
diff -ruN boost_1_49_0-boot/libs/filesystem/v3/src/operations.cpp boost_1_49_0-patched/libs/filesystem/v3/src/operations.cpp
--- boost_1_49_0-boot/libs/filesystem/v3/src/operations.cpp	2012-01-28 15:40:16.000000000 +0100
+++ boost_1_49_0-patched/libs/filesystem/v3/src/operations.cpp	2012-06-27 19:19:19.269597266 +0200
@@ -81,13 +81,15 @@
     const fs::path dot_path(".");
     const fs::path dot_dot_path("..");
 #   include <sys/types.h>
-#   if !defined(__APPLE__) && !defined(__OpenBSD__)
+#   if !defined(__APPLE__) && !defined(__OpenBSD__) && !defined(__ANDROID__) && !defined(ANDROID)
 #     include <sys/statvfs.h>
 #     define BOOST_STATVFS statvfs
 #     define BOOST_STATVFS_F_FRSIZE vfs.f_frsize
 #   else
 #     ifdef __OpenBSD__
 #     include <sys/param.h>
+#     elif defined(__ANDROID__) || defined(ANDROID) // @Moss - Android messes up a bit with some headers, this one is the correct one :D
+#     include <sys/vfs.h>
 #     endif
 #     include <sys/mount.h>
 #     define BOOST_STATVFS statfs
@@ -214,7 +216,19 @@
          || ::mkdir(to.c_str(),from_stat.st_mode)!= 0))
 #   define BOOST_COPY_FILE(F,T,FailIfExistsBool)copy_file_api(F, T, FailIfExistsBool)
 #   define BOOST_MOVE_FILE(OLD,NEW)(::rename(OLD, NEW)== 0)
+#if defined(__ANDROID__) || defined(ANDROID)
+    int BOOST_RESIZE_FILE(const char *path, off_t size)
+    {
+      int result = -1;
+      int fd = open(path, O_WRONLY);
+      if (fd != -1)
+	    result = ftruncate(fd, size);
+      close(fd);
+      return result;
+    }
+#else
 #   define BOOST_RESIZE_FILE(P,SZ)(::truncate(P, SZ)== 0)
+#endif

 #   define BOOST_ERROR_NOT_SUPPORTED ENOSYS
 #   define BOOST_ERROR_ALREADY_EXISTS EEXIST
diff -ruN boost_1_49_0-boot/libs/thread/build/Jamfile.v2 boost_1_49_0-patched/libs/thread/build/Jamfile.v2
--- boost_1_49_0-boot/libs/thread/build/Jamfile.v2	2010-10-25 10:52:02 +0300
+++ boost_1_49_0-patched/libs/thread/build/Jamfile.v2	2012-11-02 11:11:06 +0200
@@ -50,7 +50,7 @@
 local rule default_threadapi ( )
 {
     local api = pthread ;
-    if [ os.name ] = "NT" { api = win32 ; }
+#    if [ os.name ] = "NT" { api = win32 ; }
     return $(api) ;
 }
